### PR TITLE
[FIXED JENKINS-40912] Return list of files from stash/unstash steps

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-40912](https://issues.jenkins-ci.org/browse/JENKINS-40912)

downstream of https://github.com/jenkinsci/workflow-api-plugin/pull/56

cc @reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/stash/UnstashStep.java [hunk 1]
```diff
@@ -70,9 +72,9 @@ public String getName() {
             this.name = name;
         }
 
-        @Override protected Void run() throws Exception {
+        @Override protected List<String> run() throws Exception {
             StashManager.unstash(getContext().get(Run.class), name, getContext().get(FilePath.class), getContext().get(TaskListener.class));
-            return null;
+            return StashManager.fileNamesForStash(getContext().get(Run.class), name);
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/stash/UnstashStep.java]: Do I understand correctly that we're extracting the files twice?  That seems... rather awkward.  Or at least something we might consider making optional and not mandatory with each run.

**Comment by jglick** [comment]
I guess the question is whether we actually want this feature. The requester did not specifically ask for programmatic processing of the list of stashed files—only for some way of diagnosing a mistake in the pattern, which could perhaps be addressed in other ways: `FINE` logging, a `verbose` flag, **Pipeline Syntax** prevalidation of a proposed pattern given an example workspace, …

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/stash/StashStep/help.html [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <div>
     Saves a set of files for use later in the same build, generally on another node/workspace.
+    The step will return a list of files that are included in the stash.
```
Comment (rsandell) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/stash/StashStep/help.html]: File names, or paths? And if paths are they relative to the current working directory or workspace or absolute?

**Comment by abayer** [comment]
This sucker is moribund in favor of #54

**Pull request closed without merging**